### DISCUSSION TITLE
With magento2, how to display facets category wise

### DISCUSSION THREAD

**Discussion by davidandersson1** [algolia/instantsearch#5686] [Q&A]
Hello,

We are using magento version 2.4.5-p1 with Algolia module latest module (version: 3.10.4)

By default, magento algolia module provided feature to add facets at Store->Configuration->Algolia Search->Instant Search Results Page->Facets(filters).

but we want to hide cetain facets on some categories since some filters such a width, might only be available on 2-3 of 5000 products in certain categories, so filter is not relevant, but in others it is absolutely crucial and must be shown.

So is there any setting somewhere in Algolia dashaboard or in magento backend where we can set "Only show this filter / facet in category: x,y or so? 

Can you please provide your view here!

Thanks,
David Andersson

**Comment by Haroenv** [comment]
This is really a question for the magento module, not for InstantSearch. What you can do is implement DynamicWidgets in your frontend, and then you can control which facets get displayed. For more info on how to configure magento, you should check the documentation or contact support.

**Comment by davidandersson1** [comment]
Thanks for your reply! but to be honest, from all directions of algolia support, we get one same answer for this like  "implement DynamicWidget" in your frontend.

but till time nobody provided any exact example except this url
https://www.algolia.com/doc/integration/magento-2/customize/custom-front-end-events/?client=php

and on above link, there is no any example like how to implement DynamicWidget in magento2 module, there are some small and easy example provided but do not know why this kind of needy examples not included in that link.
